src/model.cpp

```diff
@@ -1315,7 +1328,14 @@ bool operator ==(const Model &a, const Model &b)
             && (a.ts == b.ts)
             && (a.nmaxevent == b.nmaxevent)
             && (a.pscale == b.pscale)
+            && (a.qpositivex == b.qpositivex)
             && (a.tstart == b.tstart);
 }
 
+void Model::computeX_pos(N_Vector x) {
```

Comment:
To keep `Model` from exploding we could make that a free function. Also improves testability.

Reply:
Ah, we want this to be a member function so we can preallocate `x_pos`.

Reply:
Can make it a free function and pass the preallocated array as second argument.